core/processor/abci.go

```diff
@@ -718,10 +722,21 @@ func (app *App) OnEndBlock(req tmtypes.RequestEndBlock) (ctx context.Context, re
 	app.stateVar.OnBlockEnd(app.blockCtx)
 
 	powerUpdates := app.top.GetValidatorPowerUpdates()
+	resp = tmtypes.ResponseEndBlock{}
 	if len(powerUpdates) > 0 {
-		resp = tmtypes.ResponseEndBlock{
-			ValidatorUpdates: powerUpdates,
-		}
+		resp.ValidatorUpdates = powerUpdates
+	}
+
+	// update max gas based on the network parameter
+
+	resp.ConsensusParamUpdates = &tmtypes.ConsensusParams{
+		Block: &tmtypes.BlockParams{
+			MaxGas:   int64(app.gastimator.OnBlockEnd()),
+			MaxBytes: 22020096,
```

Comment:
Why that hardcode default?

Reply:
it has to be something, what do you recommend?

Reply:
well, it doesn't matter so much what the average size of blocks in testnet is, but the consensus params are: 

<img width="298" alt="image" src="https://user-images.githubusercontent.com/6655556/201689053-16fba3b4-5fc6-4189-83b5-86b84a2a69aa.png">

Reply:
ah fair enough then.

Reply:
Could you use `tmtypesint.DefaultBlockParams().MaxBytes` to get the default value that tendermint sets? It'll avoid the hard-coding and make it clearer its just a default.

Reply:
yeah, that makes sense